trin-core/src/utp/packets.rs

```diff
@@ -113,90 +99,82 @@ pub struct PacketHeader {
     ack_nr: u16,
 }
 
 impl PacketHeader {
-    pub fn new() -> Self {
-        PacketHeader {
-            type_ver: u8::from(PacketType::Data) << 4 | VERSION,
-            extension: 0,
-            connection_id: 0,
-            timestamp_microseconds: 0,
-            timestamp_difference_microseconds: 0,
-            wnd_size: 0xf000,
-            seq_nr: 0,
-            ack_nr: 0,
-        }
-    }
-
-    pub fn encode(&mut self) -> Vec<u8> {
-        let mut buf = vec![];
-        buf.extend_from_slice(&[self.type_ver]);
-        buf.extend_from_slice(&[self.extension]);
-        buf.extend_from_slice(&self.connection_id.to_be_bytes());
-        buf.extend_from_slice(&self.timestamp_microseconds.to_be_bytes());
-        buf.extend_from_slice(&self.timestamp_difference_microseconds.to_be_bytes());
-        buf.extend_from_slice(&self.wnd_size.to_be_bytes());
-        buf.extend_from_slice(&self.seq_nr.to_be_bytes());
-        buf.extend_from_slice(&self.ack_nr.to_be_bytes());
-        buf
-    }
-
-    pub fn decode(bytes: &[u8]) -> PacketHeader {
-        PacketHeader {
-            type_ver: bytes[0],
-            extension: bytes[1],
-            connection_id: u16::from_be_bytes([bytes[2], bytes[3]]),
-            timestamp_microseconds: u32::from_be_bytes([bytes[4], bytes[5], bytes[6], bytes[7]]),
-            timestamp_difference_microseconds: u32::from_be_bytes([
-                bytes[8], bytes[9], bytes[10], bytes[11],
-            ]),
-            wnd_size: u32::from_be_bytes([bytes[12], bytes[13], bytes[14], bytes[15]]),
-            seq_nr: u16::from_be_bytes([bytes[16], bytes[17]]),
-            ack_nr: u16::from_be_bytes([bytes[18], bytes[19]]),
-        }
-    }
-
+    /// Returns the packet's version.
     pub fn get_version(&self) -> u8 {
-        self.type_ver & 0xf
+        self.type_ver & 0x0F
     }
 
-    pub fn set_version(&mut self, int: u8) {
-        self.type_ver = (self.type_ver & 0xf0) | (int & 0xf)
+    /// Sets the type of packet to the specified type.
+    pub fn set_type(&mut self, t: PacketType) {
+        let version = 0x0F & self.type_ver;
+        self.type_ver = u8::from(t) << 4 | version;
     }
 
+    /// Returns the packet's type.
     pub fn get_type(&self) -> PacketType {
         PacketType::try_from(self.type_ver >> 4).unwrap()
     }
+}
 
-    pub fn set_type(&mut self, t: PacketType) {
-        self.type_ver = (self.type_ver & 0xf) | (u8::from(t) << 4)
-    }
-
-    pub fn set_connection_id(&mut self, connection_id: u16) {
-        self.connection_id = connection_id;
-    }
+impl<'a> TryFrom<&'a [u8]> for PacketHeader {
+    // TODO: Refactor this to use anyhow crate
+    type Error = &'static str;
+    /// Reads a byte buffer and returns the corresponding packet header.
+    /// It assumes the fields are in network (big-endian) byte order,
+    /// preserving it.
+    fn try_from(buf: &[u8]) -> Result<Self, Self::Error> {
+        // Check length
+        if buf.len() < HEADER_SIZE {
```

Comment:
I don't have any *problem* with the simple comments like this one, but I don't think it adds much when the code is this easy to read.